### PR TITLE
Add how-to_use_tekton-pipelines-in-prow.md

### PR DESCRIPTION
**Pull request by Ressetkk**
This is getting started docs for anyone who wishes to use Tekton with Prow.

### CODE REVIEW

**Comment by Ressetkk** [comment]
/cc @dekiel
Please check the docs I there is something missing

File: docs/prow/how-to_use_tekton-pipelines-in-prow.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+* Pipelines can be marginally slower to build and define, than simple ProwJobs
+
+If you are thinking about building your pipeline with Tekton, then consider the following:
+* Does my workflow has to do some complex stuff?
+* Can I cover my requirement with quick Makefile step?
+
+If your pipeline just has to run some simple code tests, static checks, work directly on the code and does not require dependencies to external services,
+consider using Kubernetes-agent based ProwJob. 
+
+ProwJobs are suitable best for code-oriented tasks, whereas Tekton's Pipelines are a great way to implement a complex release or E2E pipeline with reusable tasks.
```
Comment (IwonaLanger) [inline docs/prow/how-to_use_tekton-pipelines-in-prow.md:227]:
```suggestion
ProwJobs are best suited for code-oriented tasks, whereas Tekton's Pipelines are a great way to implement a complex release or E2E pipeline with reusable tasks.
```

**Review by dekiel — CHANGES_REQUESTED** [review]

File: docs/prow/how-to_use_tekton-pipelines-in-prow.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+                        echo "Bye!"
+```
+
+## Re-usability of Tasks and Pipelines
+
+Kyma's Tekton pipelines and tasks are under [`configs/tekton/catalog` directory](../../configs/tekton/catalog). You can re-use any of the tasks and pipelines in your own ProwJob definition.
+
+### Re-using tasks
+
+Let's define a simple task that would return a "Hello World from task!" to the results field:
```
[the remark this answers is not in the document]
Comment (Ressetkk) [reply docs/prow/how-to_use_tekton-pipelines-in-prow.md:112]: Should we follow such strict guidelines with How to guides? It's not an official documentation. It's a guide for anyone to use specific functionality in their daily workflow.

[the remark this answers is not in the document]
Comment (IwonaLanger) [reply docs/prow/how-to_use_tekton-pipelines-in-prow.md:109]: > Should we follow such strict guidelines with How to guides? It's not an official documentation. It's a guide for anyone to use specific functionality in their daily workflow.

Well, I make suggestions based on the guidelines, at the end of the day you decide whether to apply them or not.

File: docs/prow/how-to_use_tekton-pipelines-in-prow.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+                    - name: echo
+                      image: alpine:edge
+                      script: |
+                        #!/bin/sh
+                        echo "Bye!"
+```
+
+## Re-usability of Tasks and Pipelines
+
+Kyma's Tekton pipelines and tasks are under [`configs/tekton/catalog` directory](../../configs/tekton/catalog). You can re-use any of the tasks and pipelines in your own ProwJob definition.
```
[the remark this answers is not in the document]
Comment (Ressetkk) [reply docs/prow/how-to_use_tekton-pipelines-in-prow.md]: In my opinion we have too much inconsistency in the repository. I designed `configs/` directory to contain all YAMLs with configuration related to all workloads that are automatically updated.
With that idea in mind I decided to go with the following approach:
```
configs/
\tekton/
 \deployments/ - it contains all deployment configurations that are related to Tekton
 \catalog
  |tasks/
  \pipelines/
```
Ultimately I would also see Prow configuration-related files in here in the future.

**Review by dekiel — APPROVED** [review]

File: docs/prow/how-to_use_tekton-pipelines-in-prow.md [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+  kyma-project/kyma:
+    - name: pull-ci-tekton-run
+      decorate: false
+      agent: tekton-pipeline
+      cluster: tekton-pipelines
+      always_run: true
+      tekton_pipeline_run_spec:
+        v1beta1:
+            pipelineRef:
+              name: prototype-pipeline
```
Comment (dekiel) [inline docs/prow/how-to_use_tekton-pipelines-in-prow.md:31]: Using here a pipelineRef, in the next section providing explanation for pipelineSpec and providing explanation for pipelineRef at the end of documentation might confuse users. I would use here a pipelineSpec and mention and explain usage of pipelineRef at the end.

**Review by dekiel — CHANGES_REQUESTED** [review]
Please discuss with the team a location for a tekton tasks. I would see them outside configs directory.

For v1alpha1 a Prow was automatically cloning repositories and providing as resources. That feature was removed for v1beta1?

Why we need a copy of git-clone task definition in our repo while it's published in tekton public hub?

File: docs/prow/how-to_use_tekton-pipelines-in-prow.md (hunk 4 above)
Comment (Ressetkk) [reply docs/prow/how-to_use_tekton-pipelines-in-prow.md:31]: The idea is to show difference between Kubernetes agent and Tekton pipeline definition in ProwJob, at this level it's not important for developer to know the inner requirements of a pipeline ProwJob.

**Comment by Ressetkk** [comment]
@dekiel resources are deprecated in v1beta1. I believe using supported features of git cloning as a task in workspace is a way to go.

It's in tekton hub, but installing it requires applying the resource into the cluster. This YAML has been added to keep consistency between Git and Tekton cluster. Once this PR gets merged I'll introduce small script that will apply all resources present in the directory to Tekton's cluster on post submit level.

**Review by IwonaLanger — APPROVED** [review]

File: docs/prow/how-to_use_tekton-pipelines-in-prow.md (hunk 4 above)
Comment (dekiel) [reply docs/prow/how-to_use_tekton-pipelines-in-prow.md:31]: I know the idea but it still can confuse reader.

**Comment by dekiel** [comment]
> @dekiel resources are deprecated in v1beta1. I believe using supported features of git cloning as a task in workspace is a way to go.
> 
> It's in tekton hub, but installing it requires applying the resource into the cluster. This YAML has been added to keep consistency between Git and Tekton cluster. Once this PR gets merged I'll introduce small script that will apply all resources present in the directory to Tekton's cluster on post submit level.

Right, I was asking if there is a feature of automatic repository cloning for v1beta1 like it was for v1alpha1, using a non deprecated resource or this is required to deal with it manually by adding a clone task in every pipeline?

**Comment by dekiel** [comment]
> Once this PR gets merged I'll introduce small script that will apply all resources present in the directory to Tekton's cluster on post submit level.

We could apply it directly from tekton public catalog without copy it to our repository.

**Comment by Ressetkk** [comment]
@dekiel So far from what I know only using git-clone task with workspace is supported. So For every task that operates on the code the dev will have to add this task to the pipeline.

It's in fact already added, but to prepare for GitOps way of applying new Tasks and Pipelines I decided to copy the YAML that we have applied on our cluster. Tasks and Pipelines will be automatically updated once there are changes in files.

File: docs/prow/how-to_use_tekton-pipelines-in-prow.md (hunk 4 above)
Comment (Ressetkk) [reply docs/prow/how-to_use_tekton-pipelines-in-prow.md:31]: Then what would be your suggestion to that?

Comment (dekiel) [reply docs/prow/how-to_use_tekton-pipelines-in-prow.md:31]:
```suggestion
            pipelineSpec:
              description: "Hello world!"
                tasks:
                  - name: hello
                     taskSpec:
                       steps:
                         - name: echo
                            image: alpine:edge
                            script: |
                              #!/bin/sh
                              echo "Hello World"
```

**Review by dekiel — CHANGES_REQUESTED** [review]
What about git-clone task definition copy in our repo?
Can we apply it directly from public tekton catalog?
What is a use case for maintaining it's exact copy in our repo?

**Comment by Ressetkk** [comment]
> What about git-clone task definition copy in our repo?
> Can we apply it directly from public tekton catalog?
> What is a use case for maintaining it's exact copy in our repo?

Keeping the exact same copy that we have in *our* Tekton cluster. So Git state is always reflected in Cluster.

**Review by dekiel — APPROVED** [review]
Copying definition of tasks available in other repositories doesn't make sense for me. At least for now I don't see a usage for that.

**Review by IwonaLanger — APPROVED** [review]

**Pull request merged**